src/main/java/uk/nhs/hee/tis/trainee/reference/api/GradeResource.java

```diff
@@ -73,6 +76,12 @@ public GradeResource(GradeService service, GradeMapper mapper) {
    */
   @PostMapping("/grade")
   public ResponseEntity<GradeDto> createGrade(@RequestBody GradeDto gradeDto) {
+    if (!validator.isValid(gradeDto)) {
+      // TODO: refactor service to take DTO instead of entity and move business logic there?
```

Comment:
I agree, it would make more sense to have the business logic in the service layer, and keep the api controller clean. Its not a huge issue at this stage, given the codebase is fairly small and easy to follow regardless, but I imagine if this component could grow it might be worth sorting this out now. If its going to be reworked anyway (the DMS -> mongo changes to the REST calls) then I'd be ok with leaving this as a TODO for that point in time.